### PR TITLE
ci: add ci bot for auto assign issue

### PR DESCRIPTION
**Pull request by parth-gr**
ci: add ci bot for auto assign issue

Now users can auto-assign issues by commenting
/assign on the issue topic

Closes: https://github.com/rook/rook/issues/10492
Signed-off-by: parth-gr <paarora@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: .github/workflows/auto-assign.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: contributor-auto-assign
```
Comment (travisn) [inline .github/workflows/auto-assign.yaml:1]: How about mentioning this feature in the developer guide?

File: .github/workflows/auto-assign.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+              if [[ ! -z $ISSUE_CURRENTLY_ASSIGNED_MESSAGE ]]; then
+              jq -n -r --arg body "$ISSUE_CURRENTLY_ASSIGNED_MESSAGE" '{body: $body}' > payload.json
+              curl -X POST -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" --data @payload.json https://api.github.com/repos/$REPO/issues/$ISSUE_NUMBER/comments
+              fi
+            fi
+          fi
+        shell: bash
+        env:
+          INPUT_MESSAGE: "Thanks for taking this issue! Let us know if you have any questions!"
+          INPUT_TRIGGER: "//assign"
```
Comment (travisn) [inline .github/workflows/auto-assign.yaml:38]: how about a single slash?
```suggestion
          INPUT_TRIGGER: "/assign"
```

**Comment by parth-gr** [comment]
//assign

**Comment by parth-gr** [comment]
//assign

**Comment by parth-gr** [comment]
//assign

**Comment by parth-gr** [comment]
//assign

File: .github/workflows/auto-assign.yaml (hunk 2 above)
Comment (parth-gr) [reply .github/workflows/auto-assign.yaml]: done

File: Documentation/Contributing/development-flow.md [hunk 3]
```diff
@@ -77,6 +77,11 @@ git remote -v
 
 Now you should have at least `origin` and `upstream` remotes. You can also add other remotes to collaborate with other contributors.
 
+### Auto assign Issue
+
+You can auto-assign any issue that is been created in Rook Upstream repo, by commenting
```
Comment (travisn) [inline Documentation/Contributing/development-flow.md:82]:
```suggestion
You can self-assign any issue that is been created in Rook upstream repo by adding
```

File: Documentation/Contributing/development-flow.md [hunk 4]
```diff
@@ -77,6 +77,11 @@ git remote -v
...
+`\assign` in as a issue comment.
```
Comment (travisn) [inline Documentation/Contributing/development-flow.md:83]: Does it just need to contain, or does it need to be the only text in the comment?
```suggestion
an issue comment with `/assign` in the body.
```

**Review by travisn — CHANGES_REQUESTED** [review]
Where can we see the result of running this action in your fork? I can't find it now.

**Comment by parth-gr** [comment]
> Where can we see the result of running this action in your fork? I can't find it now.

I made it private as it contains my GitHub token,
Some raw [logs](https://pipelines.actions.githubusercontent.com/serviceHosts/d58ae9c0-3eb5-4078-8c70-5480547bf0a9/_apis/pipelines/1/runs/74/signedlogcontent/2?urlExpires=2022-07-07T12%3A43%3A30.3777398Z&urlSigningMethod=HMACV1&urlSignature=Xaryw9Qww1qGnBEyFwdlO63N1xKQChZcBuoA5OIHFdI%3D)

**Comment by parth-gr** [comment]
> > > Where can we see the result of running this action in your fork? I can't find it now.
> > 
> > 
> > I made it private as it contains my GitHub token, Some raw [logs](https://pipelines.actions.githubusercontent.com/serviceHosts/d58ae9c0-3eb5-4078-8c70-5480547bf0a9/_apis/pipelines/1/runs/74/signedlogcontent/2?urlExpires=2022-07-07T12%3A43%3A30.3777398Z&urlSigningMethod=HMACV1&urlSignature=Xaryw9Qww1qGnBEyFwdlO63N1xKQChZcBuoA5OIHFdI%3D)
> 
> That link says it's expired. How about copying it to gist.github.com?

Can you access this https://gist.github.com/parth-gr/61baa7c7cd27c5bbf1a351b1225db308

File: .github/workflows/auto-assign.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                curl -X POST -H "Authorization: token $GITHUB_TOKEN" --data @payload.json https://api.github.com/repos/$REPO/issues/$ISSUE_NUMBER/comments
+              fi
+            fi
+          fi
+        shell: bash
+        env:
+          INPUT_MESSAGE: "Thanks for taking this issue! Let us know if you have any questions!"
+          INPUT_TRIGGER: "/assign"
+          ISSUE_CURRENTLY_ASSIGNED_MESSAGE: "The issue you are trying to assign to yourself is already assigned."
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (travisn) [inline .github/workflows/auto-assign.yaml:41]: Whose github token would this be? In your testing I see it was your token, but in the rook repo we don't have a bot account and I don't believe we can create a rook organization token. I don't want to use a personal token either.

**Review by travisn — CHANGES_REQUESTED** [review]
Did you get a chance to look at the bots mentioned by @BlaineEXE in #10492? It would be nice to find a bot that even has more capability than just assigning.

**Comment by parth-gr** [comment]
> Did you get a chance to look at the bots mentioned by @BlaineEXE in #10492? It would be nice to find a bot that even has more capability than just assigning.

Actually that bot was just assigning the issue only what was mentioned,

I didn't used any Bot I write the code for it on rook repo.

File: .github/workflows/auto-assign.yaml (hunk 5 above)
Comment (parth-gr) [reply .github/workflows/auto-assign.yaml:41]: `${{ secrets.GITHUB_TOKEN }}` is it contains your token?

Maybe we can discuss today on all call, or maybe we can merge and get it verified

**Comment by parth-gr** [comment]
@travisn github actions itself comments out https://github.com/parth-gr/auto-assign-issue/issues/3#issuecomment-1181743958

I have read more about tokens the link Blaine shared, and the token is used as repo-token which is by default created by actions and deleted at the end of the actions and we can change its access permission too.
Thanks

**Review by BlaineEXE — APPROVED** [review]
LGTM I think

**Review by travisn — APPROVED** [review]

File: Documentation/Contributing/development-flow.md [hunk 6]
```diff
@@ -77,6 +77,11 @@ git remote -v
 
 Now you should have at least `origin` and `upstream` remotes. You can also add other remotes to collaborate with other contributors.
 
+### Self assign Issue
+
+You can self-assign any issue that is been created in Rook upstream repo, by adding
+an issue comment with `/assign` in the body.
```
Comment (subhamkrai) [inline Documentation/Contributing/development-flow.md:83]:
```suggestion
You can self-assign any issue that is open and not assigned to anyone in Rook upstream repo, by adding a comment with `/assign` in the body.
```

**Review by subhamkrai — CHANGES_REQUESTED** [review]
nit

File: Documentation/Contributing/development-flow.md (hunk 6 above)
Comment (parth-gr) [reply Documentation/Contributing/development-flow.md]: Done changed the `not assigned` thing that will be really helpful

But `an issue comment` seems fine and more understandable so keep like that, 
let me if it doesn't look good.

**Review by subhamkrai — APPROVED** [review]

**Pull request merged**